src/auth/cognito/commands/index.ts

```diff
@@ -0,0 +1,162 @@
+  ConfirmSoftwareTokenRequest,
+  ConfirmUserAttributeRequest,
+  FetchAuthSessionRequest,
+  FetchUserAttributesRequest,
+  ForgetDeviceRequest,
+  ResendSignUpCodeRequest,
+  ResendUserAttributeConfirmationCodeRequest,
+  ResetPasswordRequest,
+  SetPreferredMFARequest,
+  SignInRequest,
+  SignInWithWebUIRequest,
+  SignOutRequest,
+  SignUpRequest,
+  UpdatePasswordRequest,
+  UpdateUserAttributeRequest,
+  UpdateUserAttributesRequest,
+} from "../../types/request";
+import { GetCurrentUserRequest } from "../../types/request/getCurrentUserRequest";
+import {
+  AuthSignInResult,
+  AuthSignUpResult,
+  AuthUser,
+  ResetPasswordResult,
+  SetupSoftwareTokenResult,
+  SignOutResult,
+  UpdateUserAttributesResult,
+} from "../../types/result";
+import { UpdateUserAttributeResult } from "../../types/result/updateUserAttributeResult";
+import { AuthSignInStep } from "../../types/step";
+import { CognitoUserAttributeKey, CustomAttribute } from "../types/models";
+import { CognitoMFAType } from "../types/models/cognitoMfaType";
+import {
+  CognitoConfirmSignInOptions,
+  CognitoConfirmSignUpOptions,
+  CognitoResendSignUpCodeOptions,
+  CognitoResendUserAttributeConfirmationCodeOptions,
+  CognitoResetPasswordOptions,
+  CognitoSignInOptions,
+  CognitoSignInWithWebUIOptions,
+  CognitoSignUpOptions,
+  CognitoUpdateUserAttributeOptions,
+  CognitoUpdateUserAttributesOptions,
+} from "../types/options";
+import { TOTPNamespace } from "../types/totpNameSpace";
+
+declare function signUp(
+  req: SignUpRequest<CognitoUserAttributeKey, CognitoSignUpOptions>
+): Promise<AuthSignUpResult<AuthStandardAtrributeKey | CustomAttribute>>;
+
+declare function confirmSignUp(
+  req: ConfirmSignUpRequest<CognitoConfirmSignUpOptions>
+): Promise<AuthSignUpResult<AuthStandardAtrributeKey | CustomAttribute>>;
+
+declare function resendSignUpCode(
+  req: ResendSignUpCodeRequest<CognitoResendSignUpCodeOptions>
+): Promise<AuthCodeDeliveryDetails<AuthStandardAtrributeKey | CustomAttribute>>;
+
+declare function signIn(
+  req: SignInRequest<CognitoSignInOptions>
+): Promise<AuthSignInResult<AuthStandardAtrributeKey | CustomAttribute>>;
+
+declare function confirmSignIn<NextSignInStep extends AuthSignInStep>(
+  req: ConfirmSignInRequest<
+    CognitoMFAType,
+    CognitoConfirmSignInOptions
+  >[NextSignInStep]
+): Promise<AuthSignInResult<AuthStandardAtrributeKey | CustomAttribute>>;
+
+declare function signInWithWebUI(
+  req: SignInWithWebUIRequest<CognitoSignInWithWebUIOptions>
+): Promise<AuthSignInResult>;
+
+declare function fetchAuthSession(
+  req?: FetchAuthSessionRequest
+): Promise<AuthSession>;
+declare function rememberDevice(): Promise<void>;
+declare function forgetDevice(
+  req?: ForgetDeviceRequest | undefined
+): Promise<void>;
+declare function fetchDevices(): Promise<AuthDevice[]>;
+
+const TOTP: TOTPNamespace = {
+  setup: function (): Promise<SetupSoftwareTokenResult> {
+    throw new Error("Function not implemented.");
```

Comment:
Could we omit implementation from these TOTP functions?

Reply:
done